content/guides/java/inject-extract.md

```diff
@@ -2,8 +2,142 @@
+
+```java
+Tags.SPAN_KIND.set(tracer.activeSpan(), Tags.SPAN_KIND_CLIENT);
+Tags.HTTP_METHOD.set(tracer.activeSpan(), "GET");
+Tags.HTTP_URL.set(tracer.activeSpan(), url.toString());
+tracer.inject(tracer.activeSpan().context(), Format.Builtin.HTTP_HEADERS, new RequestBuilderCarrier(requestBuilder));
+```
+Notice that a couple additional tags have been added to the `span` with some metadata about the HTTP request, following the OpenTracing Semantic Conventions.
+In this case the `carrier` is HTTP request headers object, which can be adapted to the carrier API by wrapping it in a helper class as follows:
+```java
+public class RequestBuilderCarrier implements io.opentracing.propagation.TextMap {
+    private final Request.Builder builder;
+
+    RequestBuilderCarrier(Request.Builder builder) {
+        this.builder = builder;
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, String>> iterator() {
+        throw new UnsupportedOperationException("carrier is write-only");
+    }
+
+    @Override
+    public void put(String key, String value) {
+        builder.addHeader(key, value);
+    }
+}
+```
+#### Extracting the Span’s Context from Incoming Request
+
+The logic on the client side instrumentation is similar, the only difference is that `tracer.extract` is used and the span is tagged as `span.kind=server`.
+
+```java
+public static Scope startServerSpan(Tracer tracer, javax.ws.rs.core.HttpHeaders httpHeaders,
+        String operationName) {
+    // format the headers for extraction
+    MultivaluedMap<String, String> rawHeaders = httpHeaders.getRequestHeaders();
+    final HashMap<String, String> headers = new HashMap<String, String>();
+    for (String key : rawHeaders.keySet()) {
+        headers.put(key, rawHeaders.get(key).get(0));
+    }
+
+    Tracer.SpanBuilder spanBuilder;
+    try {
+        SpanContext parentSpan = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(headers));
+        if (parentSpan == null) {
+            spanBuilder = tracer.buildSpan(operationName);
+        } else {
+            spanBuilder = tracer.buildSpan(operationName).asChildOf(parentSpan);
+        }
+    } catch (IllegalArgumentException e) {
+        spanBuilder = tracer.buildSpan(operationName);
+    }
+    return spanBuilder.withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER).startActive(true);
+}
+```
+
+In the above example, instead of using a dedicated adapter class to convert JAXRS `HttpHeaders` type into `io.opentracing.propagation.TextMap`, the headers are copied to a plain `HashMap<String, String>` and converted using a standard adapter `TextMapExtractAdapter`.
+
+## Injecting and Extracting TextMap
+The process of injecting and extracting `TextMap` is similar to that of HTTP. Given below are some elaborated code examples of injecting and extracting tracing information using `TextMap` format.
+
+```
```

Comment:
Add `java` after the triple-backtick

Reply:
And the same as a comment I left before: add a link to a real example on some repository.